tests/test_update.py

```
@@ -14,10 +14,19 @@
 
-def test_is_update_available_false_when_missing(
-    make_config_entry: Callable[..., MockConfigEntry], dummy_coordinator: MagicMock
+@pytest.mark.parametrize(
+    "coordinator_data",
+    [
+        pytest.param(None, id="missing"),
+        pytest.param({"firmware_update_info": {"status": "error"}}, id="error-status"),
+    ],
+)
+def test_is_update_available_false_for_missing_or_error_state(
+    coordinator_data: Any,
```

Ruff (0.15.14)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `coordinator_data`

(ANN401)